osu.Game/Overlays/SocialOverlay.cs

```diff
@@ -71,38 +72,37 @@ public SocialOverlay()
             Filter.Tabs.Current.ValueChanged += _ => queueUpdate();
 
             Filter.DisplayStyleControl.DisplayStyle.ValueChanged += style => recreatePanels(style.NewValue);
-            Filter.DisplayStyleControl.Dropdown.Current.ValueChanged += _ => queueUpdate();
+            Filter.DisplayStyleControl.Dropdown.Current.ValueChanged += _ => updateUsers(Users);
 
             currentQuery.BindTo(Filter.Search.Current);
             currentQuery.ValueChanged += query =>
             {
-                queryChangedDebounce?.Cancel();
+                Scheduler.CancelDelayedTasks();
 
                 if (string.IsNullOrEmpty(query.NewValue))
                     queueUpdate();
                 else
-                    queryChangedDebounce = Scheduler.AddDelayed(updateSearch, 500);
+                    Scheduler.AddDelayed(updateSearch, 500);
             };
         }
 
         private APIRequest getUsersRequest;
 
         private readonly Bindable<string> currentQuery = new Bindable<string>();
 
-        private ScheduledDelegate queryChangedDebounce;
-
         private void queueUpdate() => Scheduler.AddOnce(updateSearch);
 
+        private CancellationTokenSource loadCancellation;
+
         private void updateSearch()
         {
-            queryChangedDebounce?.Cancel();
+            Scheduler.CancelDelayedTasks();
```

Comment:
I'm hesitant about this. The scheduler could be used by `SearchableListOverlay` or `CompositeDrawable` and this would be potentially cancelling scheduled tasks that we are not responsible for.